pkg/hive/cell.go

```diff
@@ -53,6 +54,33 @@ func Invoke(fn any) *Cell {
 	return NewCell("", fx.Invoke(fn))
 }
 
+// OnStart registers a function to run on start up.
+func OnStart(fn any) *Cell {
```

Comment:
Hey look, it's generics (not really) :-p.

Reply:
Okay but really, this is a bit too much black magic for me. So, this is basically a decorator -- by doing evil reflection magic to ensure args are passed through. Cute... but...

For the sake of simplicity, what if this just took a `func()` and left it at that? Yes, it means a closure, but that is certainly more idiomatic.

Reply:
And if there's a reason I'm missing that a closure wouldn't work, can you at least document the magic?

Reply:
Yeah I got annoyed by having to do `fx.Invoke(func(lc fx.Lifecycle) { lc.Append(fx.Hook{OnStart: func(context.Context) error { thisOneThing() } } ) })` :)

Reply:
Documented and changed it to take a function that returns an error instead. Does this seem reasonable now? What we could also do, if you think it'd be preferred, is to write this as `func OnStart[T any](func(T) error) *Cell`. It'd constrain the function to a single input argument, but one could then use fx.In: `type Inputs struct { fx.In; Foo Foo; ...}`.

Reply:
Documenting the magic is fine for now. If it winds up needing changes later, we can do that.

It _is_ cute :-)